plugins/assimp/module/vtkF3DAssimpImporter.cxx

```diff
@@ -109,31 +110,46 @@ class vtkF3DAssimpImporter::vtkInternals
             aLight->mColorSpecular[0], aLight->mColorSpecular[1], aLight->mColorSpecular[2]);
           light->SetLightTypeToSceneLight();
 
+          bool add = true;
           switch (aLight->mType)
           {
-            case aiLightSourceType::aiLightSource_POINT:
-              light->PositionalOn();
-              break;
             case aiLightSourceType::aiLightSource_DIRECTIONAL:
               light->PositionalOff();
               break;
-            case aiLightSourceType::aiLightSource_AMBIENT:
-              light->PositionalOff();
+            case aiLightSourceType::aiLightSource_POINT:
+              light->SetConeAngle(90);
+              light->PositionalOn();
+              // Needed because of https://github.com/assimp/assimp/issues/4948
+              light->SetAttenuationValues(/*aLight->mAttenuationConstant*/ 1.0, aLight->mAttenuationLinear, aLight->mAttenuationQuadratic);
               break;
             case aiLightSourceType::aiLightSource_SPOT:
-              light->SetConeAngle(vtkMath::DegreesFromRadians(aLight->mAngleInnerCone));
+              if (this->Parent->GetColladaFixup())
+              {
+                // Needed because of https://github.com/assimp/assimp/issues/4949
+                light->SetConeAngle(vtkMath::DegreesFromRadians(aLight->mAngleInnerCone)/2);
+              }
+              else
+              {
+                light->SetConeAngle(vtkMath::DegreesFromRadians(aLight->mAngleOuterCone)/2);
+              }
               light->PositionalOn();
+              // Needed because of https://github.com/assimp/assimp/issues/4948
+              light->SetAttenuationValues(/*aLight->mAttenuationConstant*/ 1.0, aLight->mAttenuationLinear, aLight->mAttenuationQuadratic);
               break;
             case aiLightSourceType::aiLightSource_AREA:
+            case aiLightSourceType::aiLightSource_AMBIENT:
             default:
+              add = false;
```

Comment:
Can you just call `continue` to jump directly to the next loop iteration and drop the `add` variable?

Reply:
yup